### PR TITLE
Module framework improvements

### PR DESCRIPTION
**Pull request by snixtho**
- Adds ability to register custom value readers for commands.
- Hide Manialinks when a module is disabled.
- Commands to enable, disable and list modules ingame.
- Simple config to disable certain modules during startup.

Some tests are missing due to: https://github.com/evoesports/EvoSC-sharp/issues/164

### CODE REVIEW

File: EvoSC.sln [hunk 1]
```diff
@@ -146,14 +150,18 @@ Global
...
+		{5263B4FD-E1EB-4286-90E8-C02107C195D8}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{5263B4FD-E1EB-4286-90E8-C02107C195D8}.Release|Any CPU.Build.0 = Release|Any CPU
 		{75867550-1DE2-4FBB-9FC1-BA840984AF49}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{75867550-1DE2-4FBB-9FC1-BA840984AF49}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{75867550-1DE2-4FBB-9FC1-BA840984AF49}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{75867550-1DE2-4FBB-9FC1-BA840984AF49}.Release|Any CPU.Build.0 = Release|Any CPU
 		{B684B864-31BF-407E-9ED9-A95D32BC4FE2}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{B684B864-31BF-407E-9ED9-A95D32BC4FE2}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{B684B864-31BF-407E-9ED9-A95D32BC4FE2}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{B684B864-31BF-407E-9ED9-A95D32BC4FE2}.Release|Any CPU.Build.0 = Release|Any CPU
```
Comment (kdankert) [inline EvoSC.sln:156]: Those Release profiles should still exist here.

**Review by kdankert — APPROVED** [review]
Looks good to me. We might want to do some manual testing for this one as well, as unit tests are not covering everything.

File: src/EvoSC.Commands/ChatCommandManager.cs [hunk 2]
```diff
@@ -14,13 +18,28 @@ public class ChatCommandManager : IChatCommandManager
...
-    public ChatCommandManager(ILogger<ChatCommandManager> logger)
+    public IValueReaderManager ValueReader => _valueReader;
+    
+    public ChatCommandManager(ILogger<ChatCommandManager> logger, IPlayerManagerService playerManager)
     {
         _logger = logger;
+        _valueReader = SetupValueReader(playerManager);
+    }
+
+    private IValueReaderManager SetupValueReader(IPlayerManagerService playerManager)
```
Comment (AtomicLiquid) [inline src/EvoSC.Commands/ChatCommandManager.cs:33]: I would consider adding private methods below public. It's good to keep a clear hierarchy for public/private methods in classes.

**Review by AtomicLiquid — APPROVED** [review]

File: src/EvoSC.Commands/ChatCommandManager.cs (hunk 2 above)
Comment (snixtho) [reply src/EvoSC.Commands/ChatCommandManager.cs:33]: I usually put methods which are only used in the constructor close to the constructor so its all in the same place for easy navigation.

**Pull request merged**